src/AmbientSounds.Uwp/Services/ProtocolLaunchController.cs

```diff
@@ -14,42 +18,64 @@ public class ProtocolLaunchController
     private readonly IMixMediaPlayerService _player;
     private readonly IShareService _shareService;
     private readonly ITelemetry _telemetry;
+    private readonly INavigator _navigator;
 
     private const string AutoPlayKey = "autoPlay";
 
     public ProtocolLaunchController(
         IMixMediaPlayerService player,
         IShareService shareService,
-        ITelemetry telemetry)
+        ITelemetry telemetry,
+        INavigator navigator)
     {
         Guard.IsNotNull(player);
         Guard.IsNotNull(shareService);
         Guard.IsNotNull(telemetry);
+        Guard.IsNotNull(navigator);
 
         _player = player;
         _shareService = shareService;
         _telemetry = telemetry;
+        _navigator = navigator;
     }
 
-    public void ProcessLaunchProtocolArguments(string arguments)
+    public void ProcessShareProtocolArguments(string arguments)
     {
         var query = QueryString.Parse(arguments);
-        query.TryGetValue(AutoPlayKey, out var isAutoPlay);
 
-        if (!string.IsNullOrEmpty(isAutoPlay) && Convert.ToBoolean(isAutoPlay))
+        if (query.TryGetValue("id", out var shareId))
         {
-            // Auto play music.
-            _player.Play();
+            _ =_shareService.ProcessShareRequestAsync(shareId);
         }
     }
 
-    public void ProcessShareProtocolArguments(string arguments)
+    public async void ProcessAutoPlayProtocolArguments(string arguments)
     {
-        var query = QueryString.Parse(arguments);
+        bool minimize = false;
+        bool mini = false;
 
-        if (query.TryGetValue("id", out var shareId))
+        if (arguments.Contains("mini") && !arguments.Contains("minimize"))
         {
-            _ =_shareService.ProcessShareRequestAsync(shareId);
+            mini = true;
+        }
+        else if (arguments.Contains("minimize"))
+        {
+            minimize = true;
+        }
+        // Both can't really be true at the same time since AMini can't be minimized.
+
+        _player?.Play();
+
+        if (minimize)
+        {
+            IList<AppDiagnosticInfo> infos = await AppDiagnosticInfo.RequestInfoForAppAsync();
+            IList<AppResourceGroupInfo> resourceInfos = infos[0].GetResourceGroups();
+            await resourceInfos[0].StartSuspendAsync();
+        }
+
+        if (mini)
+        {
+            await _navigator.ToCompactOverlayAsync(CompactViewMode.Focus);
```

Comment:
Let's remove this for now. The mini feature is only designed for focusing at the moment. There's no "mini player" that just has a pause/play button anymore. I can add it again in the future, but it's not available right now.

Reply:
thank youuu, will do